### ISSUE TITLE
perl 5.8.7 Solaris Sparc compile error: no _ptr member in __FILE

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8084]
Migrated from [rt.perl.org#36977](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=36977) (status was 'resolved')

Searchable as RT36977$

**Comment by p5pRT** [comment]
### From manook@nortel.com




&nbsp;  
&nbsp;  


Hi there\,



I am trying to install perl5\.8\.7 on my machine but I get the following
compile errors&#8203;:

\`sh  cflags "optimize='\-O'" perlio\.o\`  perlio\.c

          CCCMD =  /localdisk/vahik/05septBuild/bin/gcc \-DPERL\_CORE \-c
\-m64 \-O  \-Wall

perlio\.c&#8203;: In function 'PerlIOStdio\_unread'&#8203;:

perlio\.c&#8203;:3064&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'

perlio\.c&#8203;:3072&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'

perlio\.c&#8203;: In function 'PerlIOStdio\_get\_base'&#8203;:

perlio\.c&#8203;:3177&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_base'

perlio\.c&#8203;:3178&#8203;: warning&#8203;: control reaches end of non\-void function

perlio\.c&#8203;: In function 'PerlIOStdio\_get\_bufsiz'&#8203;:

perlio\.c&#8203;:3184&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_cnt'

perlio\.c&#8203;:3184&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'

perlio\.c&#8203;:3184&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_base'

perlio\.c&#8203;:3185&#8203;: warning&#8203;: control reaches end of non\-void function

perlio\.c&#8203;: In function 'PerlIOStdio\_get\_ptr'&#8203;:

perlio\.c&#8203;:3193&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'

perlio\.c&#8203;:3194&#8203;: warning&#8203;: control reaches end of non\-void function

perlio\.c&#8203;: In function 'PerlIOStdio\_get\_cnt'&#8203;:

perlio\.c&#8203;:3200&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_cnt'

perlio\.c&#8203;:3201&#8203;: warning&#8203;: control reaches end of non\-void function

perlio\.c&#8203;: In function 'PerlIOStdio\_set\_ptrcnt'&#8203;:

perlio\.c&#8203;:3209&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'

perlio\.c&#8203;:3229&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_cnt'

make&#8203;: \*\*\* \[perlio\.o\] Error 1





\- I have a Solaris Sparc \(Blade 1500\)

\- my gcc is 4\.0\.1\, built using non\-gnu /usr/ccs/bin/ld and
/usr/ccs/bin/as

\- my make is gnu make 3\.80

\- I have to install the perl in my local disk

=== perl\-5\.8\.7/Configure \-Dprefix=/localdisk/vahik/05septBuild

\- for perl compile configuration\, I used all default values provided by
Configure\, except that I added \-m64 to prevent another compile error&#8203;:

=== the other compile error was&#8203;:

shutdown                            libperl\.a\(pp\_sys\.o\)

ld&#8203;: fatal&#8203;: Symbol referencing errors\. No output written to miniperl

collect2&#8203;: ld returned 1 exit status

make&#8203;: \*\*\* \[miniperl\] Error 1

=== I found the \-m64 solution when I did some googling\.



Here are more info&#8203;:

perl\-5\.8\.7 9&#8203;:56 238 > myconfig

Summary of my perl5 \(revision 5 version 8 subversion 7\) configuration&#8203;:

  Platform&#8203;:

    osname=solaris\, osvers=2\.8\, archname=sun4\-solaris

    uname='sunos wcars2at 5\.8 generic\_117350\-25 sun4u sparc
sunw\,sun\-blade\-1500 solaris '

    config\_args='\-e \-Dprefix=/localdisk/vahik/05septBuil \-D'

    hint=previous\, useposix=true\, d\_sigaction=define

    usethreads=undef use5005threads=undef useithreads=undef
usemultiplicity=undef

    useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef

    use64bitint=undef use64bitall=undef uselongdouble=undef

    usemymalloc=n\, bincompat5005=undef

  Compiler&#8203;:

    cc='/localdisk/vahik/05septBuild/bin/gcc'\, ccflags ='\-m64'\,

    optimize='\-O'\,

    cppflags='\-m64 \-m64'

    ccversion=''\, gccversion='4\.0\.1'\, gccosandvers=''

    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=4321

    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16

    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
lseeksize=8

    alignbytes=8\, prototype=define

  Linker and Libraries&#8203;:

    ld='ld'\, ldflags =' '

    libpth=/localdisk/vahik/05septBuild/lib

    libs= 

    perllibs= 

    libc=/lib/libc\.so\, so=so\, useshrplib=false\, libperl=libperl\.a

    gnulibc\_version=''

  Dynamic Linking&#8203;:

    dlsrc=dl\_none\.xs\, dlext=none\, d\_dlsymun=undef\, ccdlflags=''

    cccdlflags=''\, lddlflags=''



More info from configuration output&#8203;:

Guessing which symbols your C compiler and preprocessor define\.\.\.

Your C pre\-processor defines the following symbols&#8203;:

\_BIG\_ENDIAN

\_\_ELF\_\_

\_FILE\_OFFSET\_BITS

\_\_GNUC\_\_

\_\_GNUC\_MINOR\_\_

\_LARGEFILE64\_SOURCE

\_LARGEFILE\_SOURCE

\_LP64

\_\_LP64\_\_

\_\_STDC\_\_

\_\_SVR4

sparc

\_\_sparc

\_\_sparc\_\_

sun

\_\_sun

\_\_sun\_\_

\_\_svr4\_\_

unix

\_\_unix

\_\_unix\_\_




\-\- 


Vahik Manookian ESN&#8203;:
39\+51656

**Comment by p5pRT** [comment]
### From @nwc10

This is all we got&#8203;:

On Mon\, Aug 22\, 2005 at 07&#8203;:06&#8203;:10AM \-0700\, Vahik Manookian wrote&#8203;:
> \# New Ticket Created by  "Vahik Manookian" 
> \# Please include the string&#8203;:  \[perl \#36977\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=36977 >
> 
> 
> This transaction appears to have no content

Did the body of a bug report go missing?

The subject line suggests that perl failed to compile\, with some sort of error
about no \_ptr member in \_\_FILE

How did you run perl's Configure script? Did you pass it any command line
arguments to change the behaviour from the default?
If running it interactively did you accept the default answers when
prompted?

Nicholas Clark

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From manook@nortel.com

Hi Nicholas\,

Not only the original body is lost\, I don't have my email in my sent and 
trash box\, and this is the third time that I am sending this reply\. I 
think someone is mad at me :\-\)

Anyway\,

I am trying to install perl5\.8\.7 in my solaris sparc but I get the 
following errors&#8203;:
\`sh  cflags "optimize='\-O'" globals\.o\`  globals\.c
&nbsp;          CCCMD =  /localdisk/vahik/05septBuild/bin/gcc \-DPERL\_CORE \-c 
\-m64 \-O  \-Wall
\`sh  cflags "optimize='\-O'" perlio\.o\`  perlio\.c
&nbsp;          CCCMD =  /localdisk/vahik/05septBuild/bin/gcc \-DPERL\_CORE \-c 
\-m64 \-O  \-Wall
perlio\.c&#8203;: In function 'PerlIOStdio\_unread'&#8203;:
perlio\.c&#8203;:3064&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'
perlio\.c&#8203;:3072&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'
perlio\.c&#8203;: In function 'PerlIOStdio\_get\_base'&#8203;:
perlio\.c&#8203;:3177&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_base'
perlio\.c&#8203;:3178&#8203;: warning&#8203;: control reaches end of non\-void function
perlio\.c&#8203;: In function 'PerlIOStdio\_get\_bufsiz'&#8203;:
perlio\.c&#8203;:3184&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_cnt'
perlio\.c&#8203;:3184&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'
perlio\.c&#8203;:3184&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_base'
perlio\.c&#8203;:3185&#8203;: warning&#8203;: control reaches end of non\-void function
perlio\.c&#8203;: In function 'PerlIOStdio\_get\_ptr'&#8203;:
perlio\.c&#8203;:3193&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'
perlio\.c&#8203;:3194&#8203;: warning&#8203;: control reaches end of non\-void function
perlio\.c&#8203;: In function 'PerlIOStdio\_get\_cnt'&#8203;:
perlio\.c&#8203;:3200&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_cnt'
perlio\.c&#8203;:3201&#8203;: warning&#8203;: control reaches end of non\-void function
perlio\.c&#8203;: In function 'PerlIOStdio\_set\_ptrcnt'&#8203;:
perlio\.c&#8203;:3209&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'
perlio\.c&#8203;:3229&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_cnt'
make&#8203;: \*\*\* \[perlio\.o\] Error 1

\- The compoiler is gcc 4\.0\.1 buit using /usr/ccs/as and /usr/ccs/ld\.
\- the make is a gnu make 3\.80
\- I have to install the perl in my localdisk
\- I faced with another compile error earlier\. I found a solution for it 
by goolging\. That compile error was something to do with 64 bit 
detection\. The solution was to use \-m64\.

More info from the configuration output&#8203;:


---
Guessing which symbols your C compiler and preprocessor define\.\.\.
Your C pre\-processor defines the following symbols&#8203;:
\_BIG\_ENDIAN
\_\_ELF\_\_
\_FILE\_OFFSET\_BITS
\_\_GNUC\_\_
\_\_GNUC\_MINOR\_\_
\_LARGEFILE64\_SOURCE
\_LARGEFILE\_SOURCE
\_LP64
\_\_LP64\_\_
\_\_STDC\_\_
\_\_SVR4
sparc
\_\_sparc
\_\_sparc\_\_
sun
\_\_sun
\_\_svr4\_\_
unix
\_\_unix
\_\_unix\_\_
\_\_sun\_\_


Info from myconfig&#8203;:


---
perl\-5\.8\.7 9&#8203;:56 238 > myconfig
Summary of my perl5 \(revision 5 version 8 subversion 7\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=solaris\, osvers=2\.8\, archname=sun4\-solaris
&nbsp;    uname='sunos wcars2at 5\.8 generic\_117350\-25 sun4u sparc 
sunw\,sun\-blade\-1500 solaris '
&nbsp;    config\_args='\-e \-Dprefix=/localdisk/vahik/05septBuil \-D'
&nbsp;    hint=previous\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=undef use5005threads=undef useithreads=undef 
usemultiplicity=undef
&nbsp;    useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
&nbsp;    use64bitint=undef use64bitall=undef uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='/localdisk/vahik/05septBuild/bin/gcc'\, ccflags ='\-m64'\,
&nbsp;    optimize='\-O'\,
&nbsp;    cppflags='\-m64 \-m64'
&nbsp;    ccversion=''\, gccversion='4\.0\.1'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=4321
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, 
lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='ld'\, ldflags =' '
&nbsp;    libpth=/localdisk/vahik/05septBuild/lib
&nbsp;    libs=
&nbsp;    perllibs=
&nbsp;    libc=/lib/libc\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_none\.xs\, dlext=none\, d\_dlsymun=undef\, ccdlflags=''
&nbsp;    cccdlflags=''\, lddlflags=''

Vahik Manookian ESN&#8203;: 39\+51656

Vahik Manookian ESN&#8203;: 39\+51656


Nicholas Clark via RT wrote&#8203;:

>This is all we got&#8203;:
>
>On Mon\, Aug 22\, 2005 at 07&#8203;:06&#8203;:10AM \-0700\, Vahik Manookian wrote&#8203;:
>  
>
>>\# New Ticket Created by  "Vahik Manookian" 
>>\# Please include the string&#8203;:  \[perl \#36977\]
>>\# in the subject line of all future correspondence about this issue\. 
>>\# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=36977 >
>>
>>
>>This transaction appears to have no content
>>    
>>
>
>Did the body of a bug report go missing?
>
>The subject line suggests that perl failed to compile\, with some sort of error
>about no \_ptr member in \_\_FILE
>
>How did you run perl's Configure script? Did you pass it any command line
>arguments to change the behaviour from the default?
>If running it interactively did you accept the default answers when
>prompted?
>
>Nicholas Clark
>
>
>
>
>  
>

**Comment by p5pRT** [comment]
### From @doughera88

On Mon\, 22 Aug 2005\, Vahik Manookian wrote&#8203;:

> I am trying to install perl5\.8\.7 in my solaris sparc but I get the following
> errors&#8203;:

> \`sh  cflags "optimize='\-O'" perlio\.o\`  perlio\.c
>          CCCMD =  /localdisk/vahik/05septBuild/bin/gcc \-DPERL\_CORE \-c \-m64 \-O
> \-Wall
> perlio\.c&#8203;: In function 'PerlIOStdio\_unread'&#8203;:
> perlio\.c&#8203;:3064&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'
> perlio\.c&#8203;:3072&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'

> \- The compoiler is gcc 4\.0\.1 buit using /usr/ccs/as and /usr/ccs/ld\.
> \- the make is a gnu make 3\.80
> \- I have to install the perl in my localdisk
> \- I faced with another compile error earlier\. I found a solution for it by
> goolging\. That compile error was something to do with 64 bit detection\. The
> solution was to use \-m64\.

While that may have solved your previous unspecified problem\, I suspect it 
caused this problem\, and will cause additional ones down the road\.

Specifically\, it looks like much of Configure ran assuming a 32\-bit 
environment \(note the longsize=4 in your \./myconfig output\) but at some 
point\, you told gcc to use a 64\-bit environment \(adding in the \-m64\)\.  
This change is quite likely to invalidate many of Configure's guesses\.

I haven't tried with gcc\-4\, but perl\-5\.8\.7 should build fine on Solaris 8 
either in a 32\-bit or 64\-bit mode\, but it's best to tell Configure up 
front which you want\.  See the section

&nbsp;    =head3 64 bit support

in the INSTALL file\.

If\, instead\, you don't necessarily want 64\-bit support\, but ended up going 
down that road to try to solve that unspecified earlier problem\, then 
please let us know what that problem was and perhaps we can help solve it\.

In any case\, you should do a 

&nbsp;    make distclean

before starting over\.

> Info from myconfig&#8203;:
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> perl\-5\.8\.7 9&#8203;:56 238 > myconfig
> Summary of my perl5 \(revision 5 version 8 subversion 7\) configuration&#8203;:
>  Platform&#8203;:
>    osname=solaris\, osvers=2\.8\, archname=sun4\-solaris
>    uname='sunos wcars2at 5\.8 generic\_117350\-25 sun4u sparc sunw\,sun\-blade\-1500
> solaris '
>    config\_args='\-e \-Dprefix=/localdisk/vahik/05septBuil \-D'
>    hint=previous\, useposix=true\, d\_sigaction=define
>    usethreads=undef use5005threads=undef useithreads=undef
> usemultiplicity=undef
>    useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
>    use64bitint=undef use64bitall=undef uselongdouble=undef
>    usemymalloc=n\, bincompat5005=undef
>  Compiler&#8203;:
>    cc='/localdisk/vahik/05septBuild/bin/gcc'\, ccflags ='\-m64'\,
>    optimize='\-O'\,
>    cppflags='\-m64 \-m64'
>    ccversion=''\, gccversion='4\.0\.1'\, gccosandvers=''
>    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=4321
>    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
>    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
> lseeksize=8
>    alignbytes=8\, prototype=define
>  Linker and Libraries&#8203;:
>    ld='ld'\, ldflags =' '
>    libpth=/localdisk/vahik/05septBuild/lib
>    libs=
>    perllibs=
>    libc=/lib/libc\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
>    gnulibc\_version=''
>  Dynamic Linking&#8203;:
>    dlsrc=dl\_none\.xs\, dlext=none\, d\_dlsymun=undef\, ccdlflags=''
>    cccdlflags=''\, lddlflags=''


Hope this helps\,

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From manook@nortel.com

Thanks Andy\,

I did a make distclean and a \./Configure \-d 
\-Dprefix=/localdisk/vahik/05septBuild and this time the compile passed 
without any problem\. Thanks for the help\.

However when I installed \(make install\) the perl\, I did not get perl5 
executable\. When I checked lib directory\, I have a perl5 directory 
created there but in the bin directory I only have perl and perl5\.8\.7\.

How do I get the perl5?
Is perl5 a different application than perl?

In my system per5 \-> perl5\.00503\.

I read the README\.sloris and it looks \(to me\) perl5 is the earlier 
version of perl version 5\. I tried man perl5 but there are no entries\. 
When I do a perl5 \-help and perl \-help\, they both give very similar options\.

I am asking these since I am trying to build mozilla and it wants a full 
perl installation&#8203;:
checking whether ln \-s works\.\.\. yes
checking for perl5\.\.\. /opt/tools/bin/perl5
checking for minimum required perl version >= 5\.004\.\.\. 5\.00503
checking for full perl installation\.\.\. no
configure&#8203;: error&#8203;: Cannot find Config\.pm or $Config\{archlib\}\.  A full 
perl installation is required\.

Any help is most appreciated\.
Thanks\.

Vahik Manookian ESN&#8203;: 39\+51656

Andy Dougherty via RT wrote&#8203;:

>On Mon\, 22 Aug 2005\, Vahik Manookian wrote&#8203;:
>
>  
>
>>I am trying to install perl5\.8\.7 in my solaris sparc but I get the following
>>errors&#8203;:
>>    
>>
>>\`sh  cflags "optimize='\-O'" perlio\.o\`  perlio\.c
>>         CCCMD =  /localdisk/vahik/05septBuild/bin/gcc \-DPERL\_CORE \-c \-m64 \-O
>>\-Wall
>>perlio\.c&#8203;: In function 'PerlIOStdio\_unread'&#8203;:
>>perlio\.c&#8203;:3064&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'
>>perlio\.c&#8203;:3072&#8203;: error&#8203;: 'struct \_\_FILE' has no member named '\_ptr'
>>    
>>
>>\- The compoiler is gcc 4\.0\.1 buit using /usr/ccs/as and /usr/ccs/ld\.
>>\- the make is a gnu make 3\.80
>>\- I have to install the perl in my localdisk
>>\- I faced with another compile error earlier\. I found a solution for it by
>>goolging\. That compile error was something to do with 64 bit detection\. The
>>solution was to use \-m64\.
>>    
>>
>
>While that may have solved your previous unspecified problem\, I suspect it 
>caused this problem\, and will cause additional ones down the road\.
>
>Specifically\, it looks like much of Configure ran assuming a 32\-bit 
>environment \(note the longsize=4 in your \./myconfig output\) but at some 
>point\, you told gcc to use a 64\-bit environment \(adding in the \-m64\)\.  
>This change is quite likely to invalidate many of Configure's guesses\.
>
>I haven't tried with gcc\-4\, but perl\-5\.8\.7 should build fine on Solaris 8 
>either in a 32\-bit or 64\-bit mode\, but it's best to tell Configure up 
>front which you want\.  See the section
>
>    =head3 64 bit support
>
>in the INSTALL file\.
>
>If\, instead\, you don't necessarily want 64\-bit support\, but ended up going 
>down that road to try to solve that unspecified earlier problem\, then 
>please let us know what that problem was and perhaps we can help solve it\.
>
>In any case\, you should do a 
>
>    make distclean
>
>before starting over\.
>
>  
>
>>Info from myconfig&#8203;:
>>\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
>>perl\-5\.8\.7 9&#8203;:56 238 > myconfig
>>Summary of my perl5 \(revision 5 version 8 subversion 7\) configuration&#8203;:
>> Platform&#8203;:
>>   osname=solaris\, osvers=2\.8\, archname=sun4\-solaris
>>   uname='sunos wcars2at 5\.8 generic\_117350\-25 sun4u sparc sunw\,sun\-blade\-1500
>>solaris '
>>   config\_args='\-e \-Dprefix=/localdisk/vahik/05septBuil \-D'
>>   hint=previous\, useposix=true\, d\_sigaction=define
>>   usethreads=undef use5005threads=undef useithreads=undef
>>usemultiplicity=undef
>>   useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
>>   use64bitint=undef use64bitall=undef uselongdouble=undef
>>   usemymalloc=n\, bincompat5005=undef
>> Compiler&#8203;:
>>   cc='/localdisk/vahik/05septBuild/bin/gcc'\, ccflags ='\-m64'\,
>>   optimize='\-O'\,
>>   cppflags='\-m64 \-m64'
>>   ccversion=''\, gccversion='4\.0\.1'\, gccosandvers=''
>>   intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=4321
>>   d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
>>   ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
>>lseeksize=8
>>   alignbytes=8\, prototype=define
>> Linker and Libraries&#8203;:
>>   ld='ld'\, ldflags =' '
>>   libpth=/localdisk/vahik/05septBuild/lib
>>   libs=
>>   perllibs=
>>   libc=/lib/libc\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
>>   gnulibc\_version=''
>> Dynamic Linking&#8203;:
>>   dlsrc=dl\_none\.xs\, dlext=none\, d\_dlsymun=undef\, ccdlflags=''
>>   cccdlflags=''\, lddlflags=''
>>    
>>
>
>
>Hope this helps\,
>
>  
>

**Comment by p5pRT** [comment]
### From @doughera88

On Tue\, 23 Aug 2005\, Vahik Manookian wrote&#8203;:

> Thanks Andy\,
> 
> I did a make distclean and a \./Configure \-d
> \-Dprefix=/localdisk/vahik/05septBuild and this time the compile passed without
> any problem\. Thanks for the help\.
> 
> However when I installed \(make install\) the perl\, I did not get perl5
> executable\. When I checked lib directory\, I have a perl5 directory created
> there but in the bin directory I only have perl and perl5\.8\.7\.
> 
> How do I get the perl5?
> Is perl5 a different application than perl?

No\.  There is no standard "perl5" application\.  There is "perl"\, and there 
are specific versions\, such as "perl5\.8\.7"\, but the standard perl 
distribution doesn't install anything named 'perl5'\.

> In my system per5 \-> perl5\.00503\.

As far as I know\, by default on Solaris\, there is nothing installed by the 
name 'perl5'\.  Someone\, somewhere\, must have made it\. I can't say what 
that 'perl5' thing you have is\.  The standard perl distribution doesn't 
include it\.

> I am asking these since I am trying to build mozilla and it wants a full perl
> installation&#8203;:
> checking whether ln \-s works\.\.\. yes
> checking for perl5\.\.\. /opt/tools/bin/perl5
> checking for minimum required perl version >= 5\.004\.\.\. 5\.00503
> checking for full perl installation\.\.\. no
> configure&#8203;: error&#8203;: Cannot find Config\.pm or $Config\{archlib\}\.  A full perl
> installation is required\.

There appears to be a problem with /opt/tools/bin/perl5\.  You'd have to 
ask whoever installed it why they didn't give you a full perl 
installation\.

I don't know the Mozilla configure script at all\, but I'd expect that if 
you make sure your newly\-installed perl is first in your $PATH\, then it 
ought to find it just fine\.  Failing that\, you could temporarily rename 
the faulty /opt/tools/bin/perl5 file to something else and then re\-run 
Mozilla's configure\.

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
